inst/tests/tests.Rraw

```diff
@@ -18260,3 +18260,51 @@ test(2243.54, dt[, .I[j], x]$V1, c(1L, 3L), output="GForce TRUE")
 test(2243.55, dt[, .I[i], x]$V1, 1:4, output="GForce FALSE")
 test(2243.56, dt[, .I[1:2], x]$V1, 1:4, output="GForce FALSE")
 options(old)
+
+# 5885 implement frev
+d = c(NA, NaN, Inf, -Inf)
+test(2244.01, frev(c(0L, NA),       copy=TRUE), c(NA, 0L))
+test(2244.02, frev(1:3,             copy=TRUE), 3:1)
+test(2244.03, frev(d,               copy=TRUE), c(-Inf, Inf, NaN, NA))
+test(2244.04, frev(c(NA, 1, 0+2i),  copy=TRUE), c(0+2i, 1, NA))
+test(2244.05, frev(as.raw(0:1),     copy=TRUE), as.raw(1:0))
+test(2244.06, frev(NULL,            copy=TRUE), NULL)
+test(2244.07, frev(character(5),    copy=TRUE), character(5))
+test(2244.08, frev(integer(0),      copy=TRUE), integer(0))
+test(2244.09, frev(list(1, "a"),    copy=TRUE), list("a", 1))
+test(2244.11, frev(c(0L, NA),       copy=FALSE), c(NA, 0L))
+test(2244.12, frev(1:3,             copy=FALSE), 3:1)
+test(2244.13, frev(d,               copy=FALSE), c(-Inf, Inf, NaN, NA))
+test(2244.14, frev(c(NA, 1, 0+2i),  copy=FALSE), c(0+2i, 1, NA))
+test(2244.15, frev(as.raw(0:1),     copy=FALSE), as.raw(1:0))
+test(2244.16, frev(NULL,            copy=FALSE), NULL)
+test(2244.17, frev(character(5),    copy=FALSE), character(5))
+test(2244.18, frev(integer(0),      copy=FALSE), integer(0))
+test(2244.19, frev(list(1, "a"),    copy=FALSE), list("a", 1))
```

Comment:
Recommend adding a test for ALTREP vectors (e.g. `1:1e6`) and long vectors (e.g. 1:1e10).

Reply:
Tests needs to be light, therefore I would comment out those tests after running and confirming they work as expected.

Reply:
For making `1:1e10` work we would need to support ALTREP. Supporting ALTREP for `copy=FALSE` seems rather straightforward but creating new ALTREP objects seems to be a bit clunky.

```r
  if (ALTREP(x) && R_altrep_data2(x) == R_NilValue) {
    SEXP info = R_altrep_data1(x);
    R_xlen_t LENGTH = (R_xlen_t)REAL0(info)[0];
    int FIRST = (int)REAL0(info)[1];
    int INCR = (int)REAL0(info)[2];
    REAL0(info)[1] = FIRST+(LENGTH-1)*INCR;
    REAL0(info)[2] = INCR*(-1);
    R_set_altrep_data1(x, info);
    return x;
  }
```

Reply:
Altrep is easiest done at the R level (see below), but if there's currently no interface (i.e. not `is_altrep(x)` in data.table I withdraw my recommendation to include it in the test suite here.

```r
frev <- function(x) {
  if (is_altrep(x)) {
    return(last(x):first(x))
  }
....
}
```

Reply:
@HughParsonage Thats actually such a big brain move. Ty for that!

Reply:
Till now we always unpack altrep in DT.

Reply:
Till now we always unpacked altrep in DT.